src/plugins/telemetry_collection_manager/server/plugin.ts

```diff
@@ -133,8 +139,13 @@ export class TelemetryCollectionManagerPlugin
     const callCluster = config.unencrypted
       ? collection.esCluster.asScoped(request).callAsCurrentUser
       : collection.esCluster.callAsInternalUser;
+    // Scope the new elasticsearch Client appropriately and pass to the stats collection config
+    // Problem: I need a Scopeable request to pass to asScoped.
+    const esClient = config.unencrypted
+      ? collection.esClientGetter()!.asScoped(config.request).asCurrentUser
+      : collection.esClientGetter()!.asInternalUser;
```

Comment:
I'm not following. I have changed the expected type of the argument `collection`. It's
```
export interface Collection<
  CustomContext extends Record<string, any> = {},
  T extends BasicStatsPayload = BasicStatsPayload
> {
  customContext?: CustomContext;
  statsGetter: StatsGetter<CustomContext, T>;
  licenseGetter: LicenseGetter<CustomContext>;
  clusterDetailsGetter: ClusterDetailsGetter<CustomContext>;
  esCluster: ILegacyClusterClient;
  esClientGetter: () => IClusterClient | undefined;
  title: string;
}
```
Should I _not_ allow 'undefined' for `esClientGetter`?

As for the other option of adding an additional param, which function should i add that to? `getStatsCollectionConfig` already has the new client getter embedded in the `collection` and `getOptInStats` has the new client in the `config` argument.

Reply:
> Should I not allow 'undefined' for `esClientGetter`?

We are using the getter because it can be `undefined`. So we have to allow it.

What I meant in my comment is that the getter can return `undefined`, but `collection.esClientGetter()!` is telling TS that we know, by the time we hit this method, it will always return `IClusterClient`. The problem is that we don't have any means of ensuring it.

Re the additional param, I meant to that function only:

```
  private getStatsCollectionConfig(
    config: StatsGetterConfig,
    collection: Collection,
    collectionEsClient: ICluster,
    usageCollection: UsageCollectionSetup
  ): StatsCollectionConfig {
    const { start, end, request } = config;

    const callCluster = config.unencrypted
      ? collection.esCluster.asScoped(request).callAsCurrentUser
      : collection.esCluster.callAsInternalUser;
    // Scope the new elasticsearch Client appropriately and pass to the stats collection config
    const esClient = config.unencrypted
      ? collectionEsClient.asScoped(config.request).asCurrentUser
      : collectionEsClient.asInternalUser;

    return { callCluster, start, end, usageCollection, esClient };
  }
```

This will ensure other methods calling `getStatsCollectionConfig`, have to make sure `collection.esClientGetter()` returns the client before calling this method.